packages/create-instance/create-instance.js

```diff
@@ -109,10 +110,35 @@ export default function createInstance (
     component.options._base = _Vue
   }
 
+  function getRootVueProto (obj) {
+    while (obj) {
+      if (Object.getPrototypeOf(obj) === Vue.prototype) {
+        return obj
+      }
+
+      obj = Object.getPrototypeOf(obj)
+    }
+  }
+
+  function getExtendedComponent (component, instanceOptions) {
+    // extend _Vue to merge the mixins on _Vue
+    const extendedComponent = component.extend(_Vue).extend(instanceOptions)
+
+    // cache subclass constructor
+    component.options._Ctor[extendedComponent.cid] = extendedComponent
+
+    // to keep the possible overridden prototype and _Vue mocks on prototype,
+    // we need change the proto chains manually
+    // @see https://github.com/vuejs/vue-test-utils/pull/856
+    const root = getRootVueProto(extendedComponent.prototype)
+    Object.setPrototypeOf(root, _Vue.prototype)
```

Comment:
I believe this would mutate components:

```js
const CompA = Vue.extend()
const CompB = CompA.extend().extend()
const savedProto = CompA.prototype.__proto__
const rootProto = CompB.prototype.__proto__.__proto__
Object.setPrototypeOf(rootProto, null)

console.log(savedProto === CompA.prototype.__proto__) // false
```

Reply:
yes it could if you set prototype to be null.
But here we did was `Object.setPrototypeOf(root, _Vue.prototype)`
```js
const rootProto = CompB.prototype.__proto__.__proto__
Object.setPrototypeOf(rootProto, Vue.prototype)

console.log(savedProto === CompA.prototype.__proto__) // true
```

Reply:
If you set it to _Vue then I think it would still mutate the protoype, it just wouldn't be as severe?

```js
const CompA = Vue.extend()
const CompB = CompA.extend().extend()
const _Vue = Vue.extend()

const savedProto = CompA.prototype.__proto__
const rootProto = CompB.prototype.__proto__.__proto__
Object.setPrototypeOf(rootProto, _Vue.prototype)

console.log(savedProto === CompA.prototype.__proto__) // false
```

Reply:
I agree with what you concerned about, I will add an unit test and find a way to guarantee the super component immutably.

Reply:
That would be great, I don't have any suggestions unfortunately. I spent some time two weeks ago investigating, but I couldn't find a solution that didn't mutate the original components